### PR TITLE
Get VM primary mac address after VM power on

### PR DESCRIPTION
**Pull request by keirazhang**
The VM might not have MAC address after vm_create.yml, which led to vm_primary_nic_mac not set and failing to get VM IP. Changes in this PR fixes this issue.

### CODE REVIEW

File: linux/deploy_vm/deploy_vm_from_iso.yml [hunk 1]
```diff
@@ -201,6 +201,19 @@
   vars:
     vm_take_screenshot_local_path: "{{ current_test_log_folder }}"
 
+# VM might not have MAC address after creation without power-on
```
Comment (Tomorrow9) [inline linux/deploy_vm/deploy_vm_from_iso.yml:204]: "without power-on"? VM is powered on in line 131.

Comment (keirazhang) [reply linux/deploy_vm/deploy_vm_from_iso.yml:204]: When the VM's MAC address is allocated by ESXi host, it only gets the MAC after VM is powered on. In this test case, when the first time of executing vm_wait_primary_nic_mac.yml, the VM is powered off, so its mac is null until it is powered on.

File: env_setup/env_setup.yml [hunk 2]
```diff
@@ -77,6 +77,18 @@
...
+          include_tasks: vm_get_power_state.yml
+
+        - name: "Power on VM"
+          include_tasks: vm_set_power_state.yml
+          vars:
+            vm_power_state_set: "powered-on"
+          when: vm_power_state_get == "poweredOff"
+
+        - name: "Wait for VM primary network adapter has MAC address"
+          include_tasks: ../../common/vm_wait_primary_nic_mac.yml
```
Comment (Tomorrow9) [inline env_setup/env_setup.yml:91]: The task file path is wrong.

File: common/vm_wait_primary_nic_mac.yml [hunk 3]
```diff
@@ -41,7 +41,9 @@
...
 
-- name: "Display VM's primary network adapter MAC address"
-  ansible.builtin.debug:
-    msg:
-      - "VM primary network adapter MAC address: {{ vm_primary_nic_mac }}"
\ No newline at end of file
+- name: "Assert VM primary network adapter has MAC address"
+  ansible.builtin.assert:
+    that:
+      - vm_primary_nic_mac
```
Comment (Tomorrow9) [inline common/vm_wait_primary_nic_mac.yml:47]: Here need to use 'ansible.utils.hwaddr' to check it?

File: windows/deploy_vm/deploy_vm_from_ova.yml [hunk 4]
```diff
@@ -49,6 +49,9 @@
 
 - name: "Get VM info"
   include_tasks: ../../common/vm_get_vm_info.yml
+
+#- name: "Get VM's primary network adapter MAC address"
```
Comment (123lzxm) [inline windows/deploy_vm/deploy_vm_from_ova.yml:53]: Is it needed for windows ova installation?

File: common/vm_wait_primary_nic_mac.yml (hunk 3 above)
Comment (keirazhang) [reply common/vm_wait_primary_nic_mac.yml:47]: Updated. Thanks!

File: env_setup/env_setup.yml (hunk 2 above)
Comment (keirazhang) [reply env_setup/env_setup.yml]: Fixed. Thanks!

File: windows/deploy_vm/deploy_vm_from_ova.yml (hunk 4 above)
Comment (keirazhang) [reply windows/deploy_vm/deploy_vm_from_ova.yml:53]: These two lines were commented, which can be a reminder when more steps are added for Windows OVF testing in future.

**Comment by keirazhang** [comment]
new_vm=true
```
Testbed information:
+-------------------------------------------------------------------------------------------+
| Product | Version | Build    | Hostname or IP | Server Model                              |
+-------------------------------------------------------------------------------------------+
| ESXi    | 8.0.2   | 22380479 | x.x.x.x | VMware, Inc. VMware20,1                   |
|         |         |          |                | Intel(R) Xeon(R) Gold 6230R CPU @ 2.10GHz |
+-------------------------------------------------------------------------------------------+


VM information:
+----------------------------------------------------+
| Name                      | test_vm-3              |
+----------------------------------------------------+
| Guest OS Distribution     | FusionOS 22.0.4 x86_64 |
+----------------------------------------------------+
| IP                        |  x.x.x.x          |
+----------------------------------------------------+
| GUI Installed             | True                   |
+----------------------------------------------------+
| CloudInit Version         |                        |
+----------------------------------------------------+
| Hardware Version          | vmx-21                 |
+----------------------------------------------------+
| VMTools Version           |                        |
+----------------------------------------------------+
| Config Guest Id           | other4xLinux64Guest    |
+----------------------------------------------------+
| GuestInfo Guest Id        |                        |
+----------------------------------------------------+
| GuestInfo Guest Full Name |                        |
+----------------------------------------------------+
| GuestInfo Guest Family    |                        |
+----------------------------------------------------+
| GuestInfo Detailed Data   |                        |
+----------------------------------------------------+


Test Results (Total: 1, Passed: 1, Elapsed Time: 00:26:56)
+------------------------------------------------------+
| ID | Name                       | Status | Exec Time |
+------------------------------------------------------+
|  1 | deploy_vm_efi_sata_vmxnet3 | Passed | 00:26:41  |
+------------------------------------------------------+
```
new_vm=false
```
Testbed information:
+-------------------------------------------------------------------------------------------+
| Product | Version | Build    | Hostname or IP | Server Model                              |
+-------------------------------------------------------------------------------------------+
| ESXi    | 8.0.2   | 22380479 | x.x.x.x | VMware, Inc. VMware20,1                   |
|         |         |          |                | Intel(R) Xeon(R) Gold 6230R CPU @ 2.10GHz |
+-------------------------------------------------------------------------------------------+


VM information:
+----------------------------------------------------+
| Name                      | test_vm-2              |
+----------------------------------------------------+
| Guest OS Distribution     | FusionOS 22.0.4 x86_64 |
+----------------------------------------------------+
| IP                        | x.x.x.x          |
+----------------------------------------------------+
| GUI Installed             | True                   |
+----------------------------------------------------+
| CloudInit Version         | 19.4                   |
+----------------------------------------------------+
| Hardware Version          | vmx-21                 |
+----------------------------------------------------+
| VMTools Version           |                        |
+----------------------------------------------------+
| Config Guest Id           | other4xLinux64Guest    |
+----------------------------------------------------+
| GuestInfo Guest Id        |                        |
+----------------------------------------------------+
| GuestInfo Guest Full Name |                        |
+----------------------------------------------------+
| GuestInfo Guest Family    |                        |
+----------------------------------------------------+
| GuestInfo Detailed Data   |                        |
+----------------------------------------------------+


Test Results (Total: 1, Passed: 1, Elapsed Time: 00:01:27)
+----------------------------------------------+
| ID | Name               | Status | Exec Time |
+----------------------------------------------+
|  1 | check_inbox_driver | Passed | 00:01:10  |
+----------------------------------------------+

**Review by Tomorrow9 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**